Codeinwp/otter-blocks
# Form Storing Emails

Closes #1463, #575.


### Summary

- Creates a new top-level menu for Otter with Settings and Form Submissions as pages.
- All form submissions from the site are stored on the Form Submissions page.
- Each form submission is of a custom post type named `otter_form_record` and the data from the input fields, the submission date, and the form ID are saved as metadata with the key `otter_form_record_meta`.
- Each submission can be edited, deleted, or marked as read/unread. Creating a new submission is not allowed.
- Submission can be filtered by month, page, or form.
- A submission is marked as `read` when it is viewed (by clicking `View`) or edited.
- It is a pro feature, so if Otter Pro is not activated, the page will still be visible in the menu but with placeholder content as in the images below.
- Only the administrator has access to these form submissions.

### Screenshots 
<img width="800" alt="Screenshot 2023-04-30 at 00 27 54" src="https://user-images.githubusercontent.com/39873395/235325060-5912f01c-ff6e-4861-b14d-2a2888ba5043.png">

<img width="800" alt="Screenshot 2023-04-30 at 00 29 14" src="https://user-images.githubusercontent.com/39873395/235325065-de665e66-69e7-4f82-8a61-ab09d8b1b879.png">

<img width="800" alt="Screenshot 2023-04-30 at 00 27 25" src="https://user-images.githubusercontent.com/39873395/235325069-f54420ee-04a4-4ddf-ba1e-95f6dfe4b8e3.png">


----

### Test instructions

- Add form blocks in various places on the site, and even multiple forms on one page.
- Make different submissions and make sure all the features described above work as expected.
- Make sure that user roles other than `administrator` don't have access to see/edit/delete the form submissions. 



---- 

### Checklist before the final review
